src/mlpack/tests/kernel_test.cpp

```diff
@@ -23,158 +23,162 @@
+      Approx(17.0).epsilon(1e-7));
 }
 
 /**
  * Basic test of Euclidean distance.
  */
-BOOST_AUTO_TEST_CASE(EuclideanDistanceTest)
+TEST_CASE("EuclideanDistanceTest", "[KernelTest]")
 {
   arma::vec a = "1.0 3.0 5.0 7.0";
   arma::vec b = "4.0 0.0 2.0 0.0";
 
-  BOOST_REQUIRE_CLOSE(EuclideanDistance::Evaluate(a, b), sqrt(76.0), 1e-5);
-  BOOST_REQUIRE_CLOSE(EuclideanDistance::Evaluate(b, a), sqrt(76.0), 1e-5);
+  REQUIRE(EuclideanDistance::Evaluate(a, b) ==
+      Approx(sqrt(76.0)).epsilon(1e-7));
+  REQUIRE(EuclideanDistance::Evaluate(b, a) ==
+      Approx(sqrt(76.0)).epsilon(1e-7));
 }
 
 /**
  * Arbitrary test case for coverage.
  */
-BOOST_AUTO_TEST_CASE(ArbitraryCaseTest)
+TEST_CASE("ArbitraryCaseTest", "[KernelTest]")
 {
   arma::vec a = "3.0 5.0 6.0 7.0";
   arma::vec b = "1.0 2.0 1.0 0.0";
 
-  BOOST_REQUIRE_CLOSE((LMetric<3, false>::Evaluate(a, b)), 503.0, 1e-5);
-  BOOST_REQUIRE_CLOSE((LMetric<3, false>::Evaluate(b, a)), 503.0, 1e-5);
+  REQUIRE((LMetric<3, false>::Evaluate(a, b)) == Approx(503.0).epsilon(1e-7));
+  REQUIRE((LMetric<3, false>::Evaluate(b, a)) == Approx(503.0).epsilon(1e-7));
 
-  BOOST_REQUIRE_CLOSE((LMetric<3, true>::Evaluate(a, b)), 7.95284762, 1e-5);
-  BOOST_REQUIRE_CLOSE((LMetric<3, true>::Evaluate(b, a)), 7.95284762, 1e-5);
+  REQUIRE((LMetric<3, true>::Evaluate(a, b)) ==
+      Approx(7.95284762).epsilon(1e-7));
+  REQUIRE((LMetric<3, true>::Evaluate(b, a)) ==
+      Approx(7.95284762).epsilon(1e-7));
 }
 
 /**
  * Make sure two vectors of all zeros return zero distance, for a few different
  * powers.
  */
-BOOST_AUTO_TEST_CASE(LMetricZerosTest)
+TEST_CASE("LMetricZerosTest", "[KernelTest]")
 {
   arma::vec a(250);
   a.fill(0.0);
 
   // We cannot use a loop because compilers seem to be unable to unroll the loop
   // and realize the variable actually is knowable at compile-time.
-  BOOST_REQUIRE((LMetric<1, false>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<1, true>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<2, false>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<2, true>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<3, false>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<3, true>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<4, false>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<4, true>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<5, false>::Evaluate(a, a)) == 0);
-  BOOST_REQUIRE((LMetric<5, true>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<1, false>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<1, true>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<2, false>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<2, true>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<3, false>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<3, true>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<4, false>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<4, true>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<5, false>::Evaluate(a, a)) == 0);
+  REQUIRE((LMetric<5, true>::Evaluate(a, a)) == 0);
```

Comment:
```suggestion
  REQUIRE(LMetric<1, false>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<1, true>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<2, false>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<2, true>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<3, false>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<3, true>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<4, false>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<4, true>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<5, false>::Evaluate(a, a) == 0);
  REQUIRE(LMetric<5, true>::Evaluate(a, a) == 0);
```

Looks like the extra `()` around `LMetric` can be removed.